.azure-pipelines/common2/stages/analyze-code.yml

```diff
@@ -0,0 +1,47 @@
+parameters:
+  - name: dependsOn
+    type: object
+    default: []
+    displayName: 'Depends On'
+
+stages:
+  - stage: detect_common2_changes
+    displayName: 'Detect for changes in tests/common2'
+    jobs:
+    - job: check_common2_changes
+      steps:
+      - checkout: self
+      - script: |
+          git fetch origin
+          echo "Fetching changes from origin rc = $?"
+          CHANGED_FILES=$(git diff --name-only origin/master)
+          echo "Changed files:"
+          echo "$CHANGED_FILES"
+          if echo "$CHANGED_FILES" | grep -q "^tests/common2"; then
+            echo "Changes detected in tests/common2"
+            echo "##vso[task.setvariable variable=common2_changes]true"
+          else
+            echo "No changes detected in tests/common2"
+            echo "##vso[task.setvariable variable=common2_changes]false"
+          fi
+        displayName: 'Check if changes are in tests/common2'
+
+  - stage: analyze_code
+    dependsOn:
+      - detect_common2_changes
+      - ${{ parameters.dependsOn }}
+    condition: eq(dependencies.detect_common2_changes.outputs['check_common2_changes.common2_changes'], 'true')
+    displayName: 'Static Analysis for tests/common2'
+    jobs:
+    - template: /.azure-pipelines/common2/jobs/check-for-precommit.yml
+    - template: /.azure-pipelines/common2/jobs/run-precommit.yml
+    - template: /.azure-pipelines/common2/jobs/run-black.yml
+    - template: /.azure-pipelines/common2/jobs/run-mypy.yml
+    - template: /.azure-pipelines/common2/jobs/run-pylint.yml
+    - job: enforce_coverage
+      displayName: 'Enforce 80% Unit Test Coverage'
+      steps:
+      - script: |
+          pip install pytest pytest-cov
+          pytest tests/common2 --cov=tests/common2 --cov-report=xml --cov-fail-under=80
```

Comment:
How would the unittest scripts look like? Could they be collected by running usual sonic-mgmt test scripts?

Reply:
The current pipeline changes expect unit tests to reside under `tests/common2` along with the utility methods. They will be regular pytests. The pipeline has been configured to run unit tests in CI. But when regular tests are run via `run_test.sh` or `pytest` the unit tests are skipped as configured in `pytest.ini`. Does this look okay?